### PR TITLE
CI updates and tidying

### LINKED ISSUE #4: Cannot boot kube-master using boot.sh when metadata.json is empty
**Description**
I experienced an issue when I tried to run a k8s master on my mac.

I use the following versions:

- MacOS-X High Sierra 10.13
- linuxkit version 0.0 / commit: 41a4c2df108bc739897e6a6f9234e6c794ab380f
- moby version 0.0 / commit: 6ba3288963c52b0831e72f99851b565baf6498e4

**Steps to reproduce the issue:**
1. Clone the repo
2. cd to the repo root
3. run: make all
4. run: ./boot.sh

**Describe the results you received:**
I get the following exception:

```bash
linuxkit-kubernetes git/master  12s
❯ ./boot.sh
+ '[' -n '' ']'
+ mkdir -p kube-master-state
+ touch kube-master-state/metadata.json
+ '[' -n '' ']'
+ linuxkit run -networking default -cpus 2 -mem 1024 -state kube-master-state -disk size=4G -data kube-master-state/metadata.json --uefi kube-master-efi.iso
FATA[0000] Cannot write user data ISO: input buffer must be at least 1 byte in size
```
**Describe the results you expected:**
The k8s master should start.

**Additional information you deem important (e.g. issue happens only occasionally):**
I was able to resolve the issue by just adding an empty json object to the metadata.json file. I've committed the 'fix' to my fork: [https://github.com/synax/linuxkit-kubernetes/commit/56ad664bb721cb918ee6047b5b09f879f1b44ab1](url)

### PR DESCRIPTION
**Pull request by ijc**
* Remove some debug
* Bump to latest `linuxkit` binary (fixes #4)
* Moved a little bit of `.circleci/config.yml` to `Makefile`
* Simplified some commands ~~and refactored `.circleci/config.yml` to avoid repetition.~~
* Dropped moby tool, can use `linuxkit build` now.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -79,7 +73,7 @@ jobs:
       - run:
           name: Fetch binaries
           command: |
-            curl -fsSL -o /workspace/bin/linuxkit https://152-46932243-gh.circle-artifacts.com/0/linuxkit-linux-amd64
+            curl -fsSL -o /workspace/bin/linuxkit https://183-46932243-gh.circle-artifacts.com/0/linuxkit-linux-amd64
```
Comment (rn) [inline .circleci/config.yml]: since https://github.com/linuxkit/linuxkit/pull/2761 just got merged do you want to roll that in here and ditch `moby`

Comment (ijc) [reply .circleci/config.yml]: I guess I should, yes. After I figured out where I went wrong with my yaml-fu for the existing stuff ;-)

**Comment by ijc** [comment]
Yeah, it wasn't possible, force pushed without those nice cleanups.

Also updated to an even newer `linuxkit` and consequently dropped `moby`.

**Comment by ijc** [comment]
I made a [feature request](https://discuss.circleci.com/t/ability-to-factor-out-common-sequences-of-steps/18267) for being able to refactor like I wanted.

**Pull request merged**